types/oracledb/index.d.ts

```diff
@@ -925,6 +937,21 @@ declare namespace OracleDB {
          * @since 4.0
          */
         currentSchema?: string | undefined;
+        /**
+         * This read-only property is a string that specifies the Oracle Database domain name associated with the connection. This property returns the same value as the SQL expression:
+         * SELECT UPPER(VALUE) FROM V$PARAMETER WHERE NAME = 'db_domain';
+         * The above SQL expression returns NULL if the domain name is not specified. The dbDomain property returns an empty string in this case.
+         * 
+         * @since 6.3
+         */
+        readonly dbDomain?: string;
```

Comment:
It's highly suggested all optional props should've `| undefined` to support `exactOptionalPropertyTypes`. 

https://www.typescriptlang.org/tsconfig#exactOptionalPropertyTypes

Reply:
This is really interesting - weve had quite a few people raise PRs and requests direct about these | undefined; types - Ive mostly been following the original authors approach so didnt stop adding them until this PR - let me have a read through this link as maybe everyones wrong

Reply:
Exactly 💯